frontend/src/common/RichTextEditor.tsx

```diff
@@ -0,0 +1,47 @@
+import '@uiw/react-md-editor/markdown-editor.css'
+import '@uiw/react-markdown-preview/markdown.css'
+
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import dynamic from 'next/dynamic'
+import { ReactNode, useState } from 'react'
+
+const MDEditor = dynamic(() => import('@uiw/react-md-editor'), { ssr: false })
+
+export type RichTextEditorProps = {
+  onDataValueChange: (value: string) => void
+  dataValue: string
+  ariaLabel: string
+  label?: ReactNode
+}
+
+export default function RichTextEditor({ onDataValueChange, dataValue, ariaLabel, label }: RichTextEditorProps) {
+  const [hideToolbar, setHideToolbar] = useState(true)
+
+  const toggleToolbar = () => {
+    setHideToolbar(!hideToolbar)
+  }
+
+  const handleChange = (value?: string) => {
+    onDataValueChange(value || '')
+  }
+
+  return (
+    <>
+      <Box display='flex'>
+        {label}
```

Comment:
Label is an optional prop, what happens if you don't supply one?

Reply:
We added in label so that we could add words on the page such as "Description" on the new model page but we weren't sure if there were examples where we may have a RichTextEditor present on the page but not named above it, so kept it optional. Do you think this should be required (if perhaps we will always actually use a descriptive title above it)? Happy to change